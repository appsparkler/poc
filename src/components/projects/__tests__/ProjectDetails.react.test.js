import React from 'react'
import {Provider} from 'react-redux'
import ProjectDetails, {
  fetchProjectEffect,
  FetchedProjectCard, CardLoader,
  GetComponentJSX,
} from '../ProjectDetails.react'
import configureMockStore from 'redux-mock-store'
import renderer from 'react-test-renderer'
import thunk from 'redux-thunk'
import Adapter from 'enzyme-adapter-react-16'
import {configure, shallow} from 'enzyme'
configure({adapter: new Adapter()})
const mockStore = configureMockStore([thunk])
const props = {
  match: {
    params: {
      id: 3,
    },
  },
}
const store = mockStore({
  project: {
    fetchingProject: true,
    fetchedProject: {
      id: 1,
      title: 'Title 1',
      content: 'Content for project 1',
    },
  },
})


it('fetch-project-effect', () => {
  const dispatch = jest.fn()
  const props = {
    match: {
      params: {
        id: 2,
      },
    },
  }
  fetchProjectEffect({dispatch, props})
  expect(dispatch).toHaveBeenCalled()
})

it('should correctly render CardLoader', () => {
  const component = renderer.create(<CardLoader />)
  const tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})

it('should correctly render FetchProjectCard', () => {
  const fetchedProject = {
    id: 1,
    title: 'Title 1',
    content: 'Content for project 1',
  }
  const component = renderer.create(
      <FetchedProjectCard
        fetchedProject={fetchedProject}
      />
  )
  const tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})

it('should correctly render Project Details component', () => {
  const component = renderer.create(
      <Provider store={store}>
        <ProjectDetails {...props} />
      </Provider>
  )
  const tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})

it('should render the loader when data is being fetched...', () => {
  store.dispatch({type: 'FETCH_PROJECT_STARTED'})
  const props = {
    match: {
      params: {
        id: 3,
      },
    },
  }
  const component = shallow(
      <Provider store={store}>
        <ProjectDetails {...props} />
      </Provider>
  )
  expect(component.html()).toMatchSnapshot()
})

it('getComponentJSX should conditionally render', () => {
  const wrapper = shallow(<GetComponentJSX
    fetchingProject={true} />
  )
  expect(wrapper.html()).toMatchSnapshot()
  wrapper.setProps({
    fetchingProject: false,
    fetchedProject: {
      id: 1,
      title: 'Title 1',
      content: 'Content for project 1',
    },
  })
  expect(wrapper.html()).toMatchSnapshot()
})
