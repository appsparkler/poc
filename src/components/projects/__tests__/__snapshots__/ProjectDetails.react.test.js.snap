// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getComponentJSX should conditionally render 1`] = `"<div class=\\"project-details section container\\"><div class=\\"card z-depth-1\\"><div class=\\"card-content\\"><div class=\\"progress\\"><div class=\\"indeterminate\\"></div></div></div></div></div>"`;

exports[`getComponentJSX should conditionally render 2`] = `"<div class=\\"project-details section container\\"><div class=\\"card z-depth-1\\"><div><div class=\\"card-content\\"><h3 class=\\"card-title\\">Title 1</h3><p class=\\"card-body\\">Content for project 1</p></div><div class=\\"card-action grey lighten-4 grey-text\\"><div>Posted By The Net Ninja</div><div>2nd September, 2am</div></div></div></div></div>"`;

exports[`should correctly render CardLoader 1`] = `
<div
  className="card-content"
>
  <div
    className="progress"
  >
    <div
      className="indeterminate"
    />
  </div>
</div>
`;

exports[`should correctly render FetchProjectCard 1`] = `
<div>
  <div
    className="card-content"
  >
    <h3
      className="card-title"
    >
      Title 1
    </h3>
    <p
      className="card-body"
    >
      Content for project 1
    </p>
  </div>
  <div
    className="card-action grey lighten-4 grey-text"
  >
    <div>
      Posted By The Net Ninja
    </div>
    <div>
      2nd September, 2am
    </div>
  </div>
</div>
`;

exports[`should correctly render Project Details component 1`] = `
<div
  className="project-details section container"
>
  <div
    className="card z-depth-1"
  >
    <div
      className="card-content"
    >
      <div
        className="progress"
      >
        <div
          className="indeterminate"
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`should render the loader when data is being fetched... 1`] = `"<div class=\\"project-details section container\\"><div class=\\"card z-depth-1\\"><div class=\\"card-content\\"><div class=\\"progress\\"><div class=\\"indeterminate\\"></div></div></div></div></div>"`;
